workerTypes/index.d.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Context } from '@azure/functions';
+
+/**
+ * This module is shipped as a built-in part of the Azure Functions Node.js worker and is available at runtime
+ */
+declare module '@azure/functions-worker' {
+    /**
+     * Register a hook to interact with the lifecycle of Azure Functions.
+     * Hooks are executed in the order they were registered and will block execution if they throw an error
+     */
+    export function registerHook(
+        hookName: 'preInvocation',
+        callback: (context: PreInvocationContext) => void | Promise<void>
+    ): Disposable;
+    export function registerHook(
+        hookName: 'postInvocation',
+        callback: (context: PostInvocationContext) => void | Promise<void>
+    ): Disposable;
+    export function registerHook(hookName: string, callback: HookCallback): Disposable;
+
+    export type HookCallback = (context: {}) => void | Promise<void>;
+
+    /**
+     * Context on a function that is about to be executed
+     * This object will be passed to all pre and post invocation hooks and can be used to share information between hooks
```

Comment:
Right, that makes a lot of sense now. I misunderstood what was happening. I think in that case, maybe renaming this one might make it clearer? Maybe just `InvocationContext`? To me at least, `preInvocationContext` implied it would *only* be passed to pre-invocation hooks, not both.

In terms of comments, I guess the only thing would be updating the post invocation comment.

Reply:
Refactored hook contexts to be more intuitive